main.tf

```
@@ -197,11 +208,11 @@ resource "aws_eks_addon" "vpc_cni_addon" {
 
   cluster_name                = aws_eks_cluster.quortex.name
   addon_name                  = "vpc-cni"
-  addon_version               = var.cluster_addons["vpc-cni"].version
-  configuration_values        = try(var.cluster_addons["vpc-cni"].configuration_values, null)
-  preserve                    = try(var.cluster_addons["vpc-cni"].preserve, null)
-  resolve_conflicts_on_update = try(var.cluster_addons["vpc-cni"].resolve_conflicts, "OVERWRITE")
-  resolve_conflicts_on_create = try(var.cluster_addons["vpc-cni"].resolve_conflicts, "OVERWRITE")
+  addon_version               = var.vpc_cni_addon.version
+  configuration_values        = jsonencode(merge(local.vpc_cni_configuration_values, var.vpc_cni_addon.configuration_values))
+  preserve                    = try(var.vpc_cni_addon.preserve, null)
+  resolve_conflicts_on_update = try(var.vpc_cni_addon.resolve_conflicts, "OVERWRITE")
+  resolve_conflicts_on_create = try(var.vpc_cni_addon.resolve_conflicts, "OVERWRITE")
```

If you apply my previous suggestion, you could then simply do : 
```suggestion
  preserve                    =var.vpc_cni_addon.preserve
  resolve_conflicts_on_update = var.vpc_cni_addon.resolve_conflicts
  resolve_conflicts_on_create = var.vpc_cni_addon.resolve_conflicts
```